frontend/packages/console-app/src/components/modals/restore-pvc/restore-pvc-modal.tsx

```diff
@@ -0,0 +1,255 @@
+      }),
+      k8sGet(VolumeSnapshotClassModel, resource.spec.volumeSnapshotClassName),
+    ])
+      .then((obj) => {
+        setPVCData(obj?.[0]);
+        setDriver(obj?.[1]?.driver);
+        setProgress(false);
+      })
+      .catch((e) => {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching objects', e); // Not important to show user
+        setProgress(false);
+      });
+  }, [namespace, resource]);
+
+  const onlyPvcSCs = (scObj: StorageClassResourceKind) => {
+    if (driver) {
+      return scObj.provisioner.includes(driver);
+    }
+    return false;
+  };
+
+  const requestedSizeInputChange = ({ value, unit }) => {
+    setRequestedSize(value);
+    setRequestedUnit(unit);
+  };
+
+  const handleStorageClass = (updatedStorageClass) => {
+    const scName: string = updatedStorageClass?.metadata?.name || '';
+    setPVCStorageClass(scName);
+  };
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const snapshotName = getName(resource);
+    const accessModes = pvcData?.spec?.accessModes || ['ReadWriteOnce'];
+    const restorePVCTemplate: K8sResourceKind = {
+      apiVersion: PersistentVolumeClaimModel.apiVersion,
+      kind: PersistentVolumeClaimModel.kind,
+      metadata: {
+        name: restorePVCName,
+        namespace,
+      },
+      spec: {
+        storageClassName: pvcSC,
+        dataSource: {
+          name: snapshotName,
+          kind: VolumeSnapshotModel.kind,
+          apiGroup: VolumeSnapshotModel.apiGroup,
+        },
+        accessModes,
+        resources: {
+          requests: {
+            storage: `${requestedSize}${requestedUnit}`,
+          },
+        },
+      },
+    };
+
+    // eslint-disable-next-line promise/catch-or-return
+    handlePromise(k8sCreate(PersistentVolumeClaimModel, restorePVCTemplate, namespace)).then(
+      (newPVC) => {
+        close();
+        history.push(resourceObjPath(newPVC, referenceFor(newPVC)));
+      },
+    );
+  };
+
+  return (
+    <div className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle>Restore as new PVC</ModalTitle>
+      <Form onSubmit={submit} name="form">
+        <ModalBody>
+          <p>When restore action is finished a new crash-consistent PVC copy will be created.</p>
+          <FormGroup
+            label="Name"
+            isRequired
+            fieldId="pvc-name"
+            className="co-restore-pvc-modal__input"
+          >
+            <TextInput
+              isRequired
+              type="text"
+              id="pvc-name"
+              name="restore-pvc-modal__name"
+              value={restorePVCName}
+              onChange={setPVCName}
+            />
+          </FormGroup>
+          <FormGroup
+            fieldId="restore-storage-class"
+            className="co-restore-pvc-modal__input"
+            isRequired
+          >
+            {progress ? (
+              <LoadingInline />
```

Comment:
As per newer approach `<div className="skeleton-text" />` is being followed across the console.

Reply:
ack